sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/ConvertHelpers.java

```diff
@@ -78,12 +83,15 @@ public ConvertedSchemaInformation(
   public static <T> ConvertedSchemaInformation<T> getConvertedSchemaInformation(
       Schema inputSchema, TypeDescriptor<T> outputType, SchemaRegistry schemaRegistry) {
     ConvertedSchemaInformation<T> convertedSchema = null;
-    boolean toRow = outputType.equals(TypeDescriptor.of(Row.class));
-    if (toRow) {
+    if (outputType.equals(TypeDescriptor.of(Row.class))) {
       // If the output is of type Row, then just forward the schema of the input type to the
       // output.
       convertedSchema =
           new ConvertedSchemaInformation<>((SchemaCoder<T>) SchemaCoder.of(inputSchema), null);
+    } else if (outputType.equals(TypeDescriptor.of(GenericRecord.class))) {
+      convertedSchema =
+          new ConvertedSchemaInformation<T>(
+              (SchemaCoder<T>) AvroUtils.schemaCoder(AvroUtils.toAvroSchema(inputSchema)), null);
```

Comment:
Hm it seems like this should be generalized somehow so we don't need to have if blocks for converting to every class that doesn't encode schema information itself. For example we might want something similar for protobuf's DynamicMessage. Something pluggable like SchemaProvider, but for making conversions to/from other dynamic row types.

This is fine for now, but maybe drop a TODO and jira about generalizing?

Reply:
Agree, I will fill a JIRA and add TODO for this in a subsequent PR. I am going to merge this as it is because I have some pending PRs on top of these changes. Thanks!

Reply:
SGTM, thank you :+1:

Reply:
Jira created sorry I had not forgotten, just too long backlog :)
https://issues.apache.org/jira/browse/BEAM-11911